File: Python/cocinero.py
import threading
import time
import random
from datetime import datetime

class Cocinero(threading.Thread):
    def __init__(self, nombre, cocina):
        super().__init__()
        self.nombre = nombre
        self.cocina = cocina
    
    def preparar_pedido(self, pedido):
        # Simular tiempo de preparación entre 2 y 5 segundos
        tiempo_preparacion = random.uniform(2, 5)
        time.sleep(tiempo_preparacion)
        
        mensaje = f"{datetime.now()} - {self.nombre} ha preparado {pedido}"
        print(mensaje)
        
        # Registrar en el archivo log
        with self.cocina.log_lock:
            with open("log_pedidos.txt", "a", encoding="utf-8") as log:
                log.write(mensaje + "\n")
    
    def run(self):
        while True:
            # Intentar obtener un pedido con el lock
            with self.cocina.pedidos_lock:
                if not self.cocina.pedidos:
                    break  # No hay más pedidos, terminar
                pedido = self.cocina.pedidos.pop(0)  # Tomar el primer pedido
            
            # Preparar el pedido
            self.preparar_pedido(pedido)

File: Java/Cocinero.java
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

public class Cocinero extends Thread {
    private final String nombre;
    private final Cocina cocina;

    public Cocinero(String nombre, Cocina cocina) {
        this.nombre = nombre;
        this.cocina = cocina;
    }

    @Override
    public void run() {
        while (true) {
            Pedido pedido = cocina.tomarPedido();
            if (pedido == null) {
                break; // No hay más pedidos
            }
            
            prepararPedido(pedido);
        }
    }

    private void prepararPedido(Pedido pedido) {
        System.out.println(nombre + " está preparando: " + pedido);
        
        try {
            // Simulamos el tiempo de preparación (entre 1 y 3 segundos)
            Thread.sleep((long) (Math.random() * 2000 + 1000));
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            return;
        }

        // Registrar en el archivo log
        synchronized (Cocinero.class) {
            try (PrintWriter writer = new PrintWriter(new FileWriter("log_pedidos.txt", true))) {
                writer.println(nombre + " ha completado " + pedido);
            } catch (IOException e) {
                System.err.println("Error al escribir en el log: " + e.getMessage());
            }
        }

        System.out.println(nombre + " ha terminado de preparar: " + pedido);
    }
}